src/app/view/directives/dot-maxlength/dot-maxlength.directive.spec.ts

```diff
@@ -20,28 +19,42 @@ describe('DotMaxlengthDirective', () => {
 
         fixture = TestBed.createComponent(TestComponent);
         element = fixture.debugElement.query(By.css('div'));
-        directiveInstance = element.injector.get(DotMaxlengthDirective);
     });
 
-    it('should trigger event handler on paste', () => {
-        spyOn(directiveInstance, 'eventHandler');
-        element.triggerEventHandler('paste', 'Longer text example');
-        expect(directiveInstance.eventHandler).toHaveBeenCalled();
-    });
-
-    it('should trigger event handler on keypress', () => {
-        spyOn(directiveInstance, 'eventHandler');
-        element.triggerEventHandler('keypress', 'Longer text example');
-        expect(directiveInstance.eventHandler).toHaveBeenCalled();
-    });
+    it(
+        'should allow new characters when max length is not reached',
+        fakeAsync(() => {
+            const event = new Event('keypress');
+            fixture.detectChanges();
+            element.nativeElement.dispatchEvent(event);
+            element.nativeElement.textContent = element.nativeElement.textContent + 'o';
```

Comment:
Here looks like you're doing what the directive should do it it self... are you sure about this test?